src/aiokatcp/sensor.py

```diff
@@ -226,9 +226,10 @@ def notify(self, reading: Reading[_T], old_reading: Reading[_T]) -> None:
         Users should not usually call this directly. It is called automatically
         by :meth:`set_value`.
         """
-        for classic_observer in self._classic_observers:
+        # list() is to avoid errors if the set is modified during iteration.
+        for classic_observer in list(self._classic_observers):
             classic_observer(self, reading)
-        for delta_observer in self._delta_observers:
+        for delta_observer in list(self._delta_observers):
```

Comment:
Noted. Although, how does it do that? By wrapping in `list(..)`, does it e.g. 'keep the cast copy in place until we're done using it'?

Reply:
`list` makes a copy. So we're iterating over a copy (which won't change) rather than the original (which might).